src/Provider/Instagram.php

```diff
@@ -55,6 +55,85 @@ protected function initialize()
         $this->apiRequestParameters[$this->accessTokenName] = $accessToken;
     }
 
+    /**
+     * {@inheritdoc}
+     */
+    protected function validateAccessTokenExchange($response)
+    {
+        $collection = parent::validateAccessTokenExchange($response);
+
+        if (!$collection->exists('expires_in')) {
+            // Instagram tokens always expire in an hour, but this is implicit not explicit
+
+            $expires_in = 60 * 60;
+
+            $expires_at = time() + $expires_in;
+
+            $this->storeData('expires_in', $expires_in);
+            $this->storeData('expires_at', $expires_at);
+        }
+
+        return $collection;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function apiRequest($url, $method = 'GET', $parameters = [], $headers = [], $multipart = false)
```

Comment:
Assume that `maintainToken` should be called in apiRequest - instead of overriding this method in Facebook and Instagram providers. Can we call it here? https://github.com/hybridauth/hybridauth/blob/master/src/Adapter/OAuth2.php#L710